File: examples/ejemplo_con_errores.py
"""
Archivo de ejemplo con errores intencionales para demostrar el AI Code Reviewer.
NO usar este código en producción.
"""
import os
import sqlite3


# ── OWASP A02: Credenciales hardcodeadas ─────────────────────────────────────
password = "super_secret_123"
api_key = "sk-abc123xyz789realtoken"
db_token = "prod-db-token-hardcoded"


# ── OWASP A03: SQL Injection por concatenación ────────────────────────────────
def get_user_by_id(user_id):
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE id = " + user_id)
    row = cursor.fetchone()
    conn.close()
    return row


# ── OWASP A03: SQL Injection por f-string ─────────────────────────────────────
def get_user_by_email(email):
    conn = sqlite3.connect("users.db")
    cursor = conn.cursor()
    cursor.execute(f"SELECT * FROM users WHERE email = '{email}'")
    result = cursor.fetchall()
    conn.close()
    return result


# ── OWASP A03: Command injection ─────────────────────────────────────────────
def generate_thumbnail(filename):
    os.system("convert " + filename + " -resize 100x100 thumb.png")


# ── OWASP A03: eval() con input dinámico ─────────────────────────────────────
def calculate_formula(expression):
    result = eval(expression)
    return result


# ── Clean Code: Función demasiado larga + nombres de 1 letra + anidamiento profundo ──
def validate_and_process_user_data(u, p, e, r, a):
    if u:
        if len(u) > 3:
            if p:
                if len(p) > 5:
                    if "@" in e:
                        if r in ["admin", "user", "guest"]:
                            if a:
                                x = u.strip()
                                y = p.strip()
                                z = e.strip()
                                return {
                                    "username": x,
                                    "password": y,
                                    "email": z,
                                    "role": r,
                                    "active": a,
                                }
                            else:
                                return None
                        else:
                            return None
                    else:
                        return None
                else:
                    return None
            else:
                return None
        else:
            return None
    else:
        return None


# ── Complejidad ciclomática alta ──────────────────────────────────────────────
def calculate_discount(price, user_type, quantity, is_member, coupon, region, day_of_week):
    d = 0
    if user_type == "premium":
        d += 10
    elif user_type == "vip":
        d += 20
    elif user_type == "gold":
        d += 15
    if quantity > 10:
        d += 5
    elif quantity > 5:
        d += 3
    if is_member:
        d += 2
    if coupon == "SAVE10":
        d += 10
    elif coupon == "SAVE20":
        d += 20
    elif coupon == "SAVE30":
        d += 30
    if region == "south":
        d += 1
    elif region == "north":
        d += 2
    elif region == "east":
        d += 1
    if day_of_week in ["Saturday", "Sunday"]:
        d += 5
    if d > 50:
        d = 50
    return price * (1 - d / 100)

File: examples/ejemplo_con_errores.js
/**
 * Archivo de ejemplo con errores intencionales para demostrar el AI Code Reviewer.
 * NO usar este código en producción.
 */

// ── OWASP A02: Credenciales hardcodeadas ─────────────────────────────────────
const apiKey = "sk-abc123xyz789realtoken";
const password = "hardcoded_password_456";

// ── OWASP A03: SQL Injection con template literal ─────────────────────────────
async function getUserByUsername(username) {
    const query = `SELECT * FROM users WHERE username = '${username}'`;
    const result = await db.query(query);
    return result;
}

// ── OWASP A03: SQL Injection con concatenación ────────────────────────────────
function getProductById(id) {
    const query = "SELECT * FROM products WHERE id = " + id;
    return db.execute(query);
}

// ── OWASP A03: eval() con input del usuario ───────────────────────────────────
function calculateFormula(userInput) {
    return eval(userInput);
}

// ── OWASP A07: XSS via innerHTML ─────────────────────────────────────────────
function renderUserProfile(userData) {
    const container = document.getElementById("profile");
    container.innerHTML = "<h2>" + userData.name + "</h2><p>" + userData.bio + "</p>";
}

// ── Clean Code: Función larga + variables de 1 letra + complejidad alta ───────
function processOrder(o, u, p, d) {
    var x = 0;
    var y = 0;
    var z = null;

    if (o.items && o.items.length > 0) {
        for (var i = 0; i < o.items.length; i++) {
            var item = o.items[i];
            if (item.quantity > 0) {
                if (item.price > 0) {
                    if (item.available) {
                        x += item.quantity * item.price;
                        y += item.quantity;
                    }
                }
            }
        }
    }

    if (u && u.isPremium) {
        x = x * 0.9;
    } else if (u && u.isVip) {
        x = x * 0.8;
    } else if (u && u.isGold) {
        x = x * 0.85;
    }

    if (d && d.code === "SAVE10") {
        x = x * 0.9;
    } else if (d && d.code === "SAVE20") {
        x = x * 0.8;
    } else if (d && d.code === "SAVE30") {
        x = x * 0.7;
    }

    if (p === "credit_card") {
        z = "CC";
    } else if (p === "paypal") {
        z = "PP";
    } else if (p === "crypto") {
        z = "CR";
    } else {
        z = "CASH";
    }

    return {
        total: x,
        itemCount: y,
        paymentMethod: z,
        status: "processed"
    };
}
